File: ai-hub-backend/src/models/apiKeyModel.js
import mongoose from 'mongoose';

const apiKeySchema = new mongoose.Schema(
  {
    modelName: {
      type: String,
      required: true,
      trim: true,
    },
    provider: {
      type: String,
      required: true,
      trim: true,
    },
    keyName: {
      type: String,
      required: true,
      trim: true,
    },
    keyValue: {
      type: String,
      required: true,
    },
    status: {
      type: String,
      enum: ['active', 'inactive'],
      default: 'active',
    },
    balance: {
      type: Number,
      default: null,
      min: 0,
    },
    usageEnv: {
      type: String,
      enum: ['dev', 'prod', 'test'],
      default: 'prod',
    },
    ownerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Admin',
      required: true,
    },
  },
  { timestamps: true }
);

apiKeySchema.index({ keyName: 1, usageEnv: 1 }, { unique: true });

export default mongoose.models.ApiKey || mongoose.model('ApiKey', apiKeySchema);

File: backend/infra/db/mongoose/models/apikey_model.go
package models

import "time"

type APIKeyDoc struct {
	ID        string    `bson:"_id"`
	OwnerId   string    `bson:"ownerId"`
	KeyName   string    `bson:"keyName"`
	Provider  string    `bson:"provider"`
	ModelName string    `bson:"modelName"`
	KeyValue  string    `bson:"keyValue"`
	UsageEnv  string    `bson:"usageEnv"`
	Status    string    `bson:"status"`
	Balance   float64   `bson:"balance"`
	CreatedAt time.Time `bson:"createdAt"`
	UpdatedAt time.Time `bson:"updatedAt"`
}
